cuda_core/cuda/core/experimental/_strided_copy/_jit.pyx

```diff
@@ -0,0 +1,105 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from libc.stdint cimport intptr_t
+
+import os
+import threading
+
+from cuda.core.experimental._device import Device
+from cuda.core.experimental._program import Program, ProgramOptions
+
+
+_tls = threading.local()
+
+# In multithreaded environment we share the compiled and loaded modules between threads.
+# Each thread has its own cache mapping arch -> kernel_code_str -> Kernel ptr,
+# on a cache miss, we first take a look into the shared cache guarded with _kernel_lock
+# and eventually compile if needed.
+_kernel_lock = threading.Lock()
+_kernel_cache = {}  # arch -> kernel_code_str -> Kernel
+
+
+cpdef str get_strided_copy_include_dir(object logger):
+    """
+    Finds and caches the absolute path for the strided copy includes.
+    """
+    # TODO(ktokarski) Once Program API supports passing includes as strings and names,
+    # read all the headers once and cache them.
+    if hasattr(_tls, "strided_copy_include_dir"):
+        return _tls.strided_copy_include_dir
+    cdef str current_dir = os.path.dirname(os.path.abspath(__file__))
+    cdef str copy_kernel_dir = os.path.normpath(os.path.join(current_dir, "..", "include", "strided_copy"))
+    _tls.strided_copy_include_dir = copy_kernel_dir
+    if logger is not None:
+        logger.debug(f"Strided copy include dir: {copy_kernel_dir}")
+    return copy_kernel_dir
+
+
+cdef inline str get_device_arch(int device_id):
+    # device_id -> arch
+    if not hasattr(_tls, "device_ccs"):
+        _tls.device_ccs = {}
+    cdef dict device_ccs = _tls.device_ccs
+    cdef str arch = device_ccs.get(device_id)
+    if arch is None:
+        arch = f"sm_{Device(device_id).arch}"
+        device_ccs[device_id] = arch
+    return arch
+
+
+cdef compile_load_kernel(str kernel_code, str arch, object logger):
+    cdef str include_dir = get_strided_copy_include_dir(logger)
+    cdef options = ProgramOptions(std="c++17", arch=arch, include_path=include_dir)
```

Comment:
Hardcoding `c++17' feels a bit brittle here, but I certainly won't block on that.

Reply:
C++17 is the minimal starting CUDA 13, so I am fine with this.